src/ClientWidgetApi.ts

```diff
@@ -770,6 +830,8 @@ export class ClientWidgetApi extends EventEmitter {
                     return this.replyVersions(<ISupportedVersionsActionRequest>ev.detail);
                 case WidgetApiFromWidgetAction.SendEvent:
                     return this.handleSendEvent(<ISendEventFromWidgetActionRequest>ev.detail);
+                case WidgetApiFromWidgetAction.SendFuture:
+                    return this.handleSendFuture(<ISendFutureFromWidgetActionRequest>ev.detail);
```

Comment:
here we would add another condition where we call handle `handleSendFuture` or `handleSendEvent` based on `ev.detail.timeout`